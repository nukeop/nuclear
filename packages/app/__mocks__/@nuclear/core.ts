const initialStoreState = () => ({
  equalizer: {
    selected: 'Default'
  },
  downloads: [],
  favorites: {
    albums: [],
    tracks: []
  },
  playlists: []
});

let mockStore = initialStoreState();

module.exports = {
  store: {
    get: (key: string) => mockStore[key] || {},
    set: (key: string, value: any) => {
      mockStore[key] = value;
    },
    clear: () => mockStore = initialStoreState()
  },
  createApi: () => ({
    app: {},
    store: {},
    React: jest.fn(),
    ReactDOM: jest.fn()
  }),
  getOption: () => '',
  rest: {
    LastFmApi: class {
      constructor() {}

      getTagInfo() {}
      getTagTracks() {}
      getTagAlbums() {}
      getTagArtists() {}
      getTopTags = () => Promise.resolve()
      getTopTracks = () => Promise.resolve()
    }
  },
  settingsConfig: jest.requireActual('@nuclear/core/src/settings').settingsConfig,
  SettingType: {
    BOOLEAN: 'boolean',
    LIST: 'list',
    NODE: 'node',
    NUMBER: 'number',
    STRING: 'string',
    DIRECTORY: 'directory'
  }
};
