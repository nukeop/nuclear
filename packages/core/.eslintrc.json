{
  "extends": [
    "../../.eslintrc.json"
  ],
  "env": {
    "node": true,
    "browser": true,
    "commonjs": true
  },
  "rules": {
    "node/no-missing-import": 0,
    "node/no-extraneous-import": 0
  }
}
