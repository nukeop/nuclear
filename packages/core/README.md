# `@nuclear/core`

> TODO: description

## Usage

```
const core = require('@nuclear/core');

// TODO: DEMONSTRATE API
```
