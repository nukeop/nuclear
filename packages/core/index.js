export {
  LastFmApi,
  isElectron
} from './lib';
