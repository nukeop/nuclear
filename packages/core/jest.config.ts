export default async () => ({
  transform: {
    '^.+\\.(ts|tsx|js|jsx)$': 'ts-jest'
  },
  moduleNameMapper: {
    '^react$': '<rootDir>/../../node_modules/react'
  }
});
