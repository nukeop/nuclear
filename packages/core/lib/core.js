'use strict';

module.exports = core;

function core() {
  // TODO
}
