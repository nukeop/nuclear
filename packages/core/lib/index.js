import 'isomorphic-fetch';
export { default as LastFmApi } from './rest/lastfm';
