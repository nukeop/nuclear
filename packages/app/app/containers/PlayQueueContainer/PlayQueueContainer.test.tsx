import { fireEvent, waitFor } from '@testing-library/react';
import { store as electronStore } from '@nuclear/core';

import { AnyProps, mountedPlayQueueFactory, setupI18Next } from '../../../test/testUtils';
import { buildElectronStoreState, buildStoreState } from '../../../test/storeBuilders';
import userEvent from '@testing-library/user-event';

describe('Play Queue container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const { store } = require('@nuclear/core');
    store.clear();
  });

  it('should display tracks in the queue', async () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should collapse and expand the queue sidebar on click', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getByTestId('queue-menu-collapse').click());

    let state = store.getState();
    expect(state.settings.compactQueueBar).toBe(true);

    await waitFor(() => component.getByTestId('queue-menu-collapse').click());
    state = store.getState();
    expect(state.settings.compactQueueBar).toBe(false);
  });

  it('should display a context popup on right click', async () => {
    const { component } = mountComponent();
    const track = component.getByTestId('queue-popup-uuid1');
    await waitFor(() => fireEvent.contextMenu(track));
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display a context popup on right click for tracks without thumbnails', async () => {
    const { component } = mountComponent();
    const track = component.getByTestId('queue-popup-uuid3');
    await waitFor(() => fireEvent.contextMenu(track));
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should select a new stream from the popup', async () => {
    const { component, store } = mountComponent();
    const track = component.getByTestId('queue-popup-uuid1');
    await waitFor(() => fireEvent.contextMenu(track));

    await waitFor(() => component.getByText(/Different Stream Provider/i).click());
    const state = store.getState();
    expect(state.queue.queueItems[0].stream.source).toBe('Different Stream Provider');
  });

  it('should copy the original track url to clipboard', async () => {
    const { component } = mountComponent();
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const clipboard = require('electron').clipboard;

    const track = component.getByTestId('queue-popup-uuid1');
    await waitFor(() => fireEvent.contextMenu(track));
    await waitFor(() => component.getByTestId('copy-original-url').click());
    expect(clipboard.writeText).toHaveBeenCalledWith('https://test-track-original-url');
  });

  it('should not display the copy original track url button if the url is not included in the current stream', async () => {
    const { component } = mountComponent();
    const track = component.getByTestId('queue-popup-uuid2');
    await waitFor(() => fireEvent.contextMenu(track));
    const copyButton = component.queryByTestId('copy-original-url');
    expect(copyButton).toEqual(null);
  });

  it('should favorite track without stream data (track queue popup)', async () => {
    const { component, store } = mountComponent();

    const track = component.getByTestId('queue-popup-uuid1');
    await waitFor(() => fireEvent.contextMenu(track));
    await waitFor(() => component.getByTestId('queue-popup-favorite').click());

    const state = store.getState();
    expect(state.favorites.tracks).toEqual(expect.arrayContaining([{
      uuid: expect.any(String),
      artist: expect.stringMatching('test artist 1'),
      name: expect.stringMatching('test track 1'),
      thumbnail: 'https://test-track-thumb-url'
    }]));

    expect(state.favorites.tracks[0].stream).toBeUndefined();
  });

  it('should favorite track without stream data (queue menu popup)', async () => {
    const { component, store } = mountComponent();

    await waitFor(() => component.getByTestId('queue-menu-more-container').click());
    await waitFor(() => component.getByTestId('queue-menu-more-favorite').click());

    const state = store.getState();
    expect(state.favorites.tracks).toEqual(expect.arrayContaining([{
      uuid: expect.any(String),
      artist: expect.stringMatching('test artist 1'),
      name: expect.stringMatching('test track 1'),
      thumbnail: 'https://test-track-thumb-url'
    }]));

    expect(state.favorites.tracks[0].stream).toBeUndefined();
  });

  it('should add the current track to a playlist (queue menu popup)', async () => {
    const { component, store } = mountComponent();

    await waitFor(() => component.getByTestId('queue-menu-more-container').click());
    await waitFor(() => component.getByText(/Add to playlist/i).click());
    await waitFor(() => component.getByText('test playlist').click());

    const state = store.getState();

    expect(state.playlists.localPlaylists.data[0].tracks).toEqual([
      expect.objectContaining({
        artist: 'test artist 1',
        name: 'test track 1'
      })
    ]);
  });

  it('should create a new playlist from the queue', async () => {
    const { component, store } = mountComponent();

    await waitFor(() => component.getByTestId('queue-menu-more-container').click());
    await waitFor(() => component.getByText(/Save as playlist/i).click());
    userEvent.type(
      component.getByPlaceholderText('Playlist name...'), 
      '{selectall}{backspace}my  new playlist{enter}'
    );

    const state = store.getState();

    expect(state.playlists.localPlaylists.data[1].name).toBe('my new playlist');
    expect(state.playlists.localPlaylists.data[1].tracks).toEqual([
      expect.objectContaining({
        artist: 'test artist 1',
        name: 'test track 1'
      }), 
      expect.objectContaining({
        artist: 'test artist 2',
        name: 'test track 2'
      }),
      expect.objectContaining({
        artist: 'test artist 3',
        name: 'test track 3'
      })
    ]);
  });

  it('should clear the queue', async () => {
    const { component, store } = mountComponent();

    await waitFor(() => component.getByTestId('queue-menu-more-container').click());
    await waitFor(() => component.getByText(/Clear queue/i).click());

    const state = store.getState();
    expect(state.queue.queueItems).toEqual([]);
  });

  const mountComponent = (initialStore?: AnyProps) => {
    const initialState = initialStore || buildStoreState()
      .withTracksInPlayQueue()
      .withPlaylists([{
        id: 'test-playlist-id',
        name: 'test playlist',
        tracks: []
      }])
      .withPlugins()
      .withConnectivity()
      .build();

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    electronStore.init({
      ...buildElectronStoreState(),
      playlists: initialState.playlists.localPlaylists.data
    });
    
    return mountedPlayQueueFactory(
      ['/dashboard'],
      initialState
    )();
  };
});
