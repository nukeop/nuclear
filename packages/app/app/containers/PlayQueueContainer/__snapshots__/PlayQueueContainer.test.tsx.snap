// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play Queue container should display a context popup on right click 1`] = `
<DocumentFragment>
  <div
    class="play_queue_container"
  >
    <div
      class="queue_menu_container"
    >
      <div
        class="queue_menu_buttons"
      >
        <a
          class="compactButton"
          data-testid="queue-menu-collapse"
          href="#"
        >
          <i
            aria-hidden="true"
            class="angle right icon"
          />
        </a>
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="ui item dropdown queue_menu_more"
          data-testid="queue-menu-more-container"
          role="listbox"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="ellipsis vertical icon"
          />
          <div
            class="menu transition"
          >
            <div
              class="header"
            >
              Queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="trash icon"
              />
              Clear queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="save icon"
              />
              Save as playlist
            </div>
            <div
              class="divider"
            />
            <div
              class="header"
            >
              Current track
            </div>
            <div
              class="item"
              role="option"
            >
              <div
                aria-expanded="false"
                class="ui dropdown left"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  class="divider text"
                  role="alert"
                >
                  Add to playlist
                </div>
                <i
                  aria-hidden="true"
                  class="dropdown icon"
                />
                <div
                  class="menu transition left playlists_menu"
                >
                  <div
                    class="item"
                    role="option"
                  >
                    <i
                      aria-hidden="true"
                      class="music icon"
                    />
                    test playlist
                  </div>
                </div>
              </div>
            </div>
            <div
              class="item"
              data-testid="queue-menu-more-favorite"
              role="option"
            >
              <i
                aria-hidden="true"
                class="star icon"
              />
              Add to favorites
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
              Download
            </div>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      class="play_queue_items fade_in"
      data-rbd-droppable-context-id="2"
      data-rbd-droppable-id="play_queue"
    >
      <div
        aria-describedby="rbd-hidden-text-2-hidden-text-10"
        data-rbd-drag-handle-context-id="2"
        data-rbd-drag-handle-draggable-id="uuid1+0"
        data-rbd-draggable-context-id="2"
        data-rbd-draggable-id="uuid1+0"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid1"
        >
          <div
            class="nuclear queue_item current_song"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 1
              </div>
              <div
                class="artist_container"
              >
                test artist 1
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-2-hidden-text-10"
        data-rbd-drag-handle-context-id="2"
        data-rbd-drag-handle-draggable-id="uuid2+1"
        data-rbd-draggable-context-id="2"
        data-rbd-draggable-id="uuid2+1"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid2"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 2
              </div>
              <div
                class="artist_container"
              >
                test artist 2
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-2-hidden-text-10"
        data-rbd-drag-handle-context-id="2"
        data-rbd-drag-handle-draggable-id="uuid3+2"
        data-rbd-draggable-context-id="2"
        data-rbd-draggable-id="uuid3+2"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid3"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="test-file-stub"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 3
              </div>
              <div
                class="artist_container"
              >
                test artist 3
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-2"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-2-hidden-text-10"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="ui bottom center popup transition visible queue_popup hidden"
      style="position: initial;"
    >
      <div
        class="stream_info"
      >
        <div
          class="stream_thumbnail"
        />
        <div
          class="stream_text_info"
        >
          <div
            class="stream_source"
          >
            <label>
              Source:
            </label>
             
            <div
              aria-expanded="false"
              class="ui inline dropdown"
              data-testid="stream-info-dropdown"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-atomic="true"
                aria-live="polite"
                class="divider text"
                role="alert"
              >
                Test Stream Provider
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              >
                <div
                  aria-checked="true"
                  aria-selected="true"
                  class="active selected item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    Test Stream Provider
                  </span>
                </div>
                <div
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    Different Stream Provider
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="stream_title"
          >
            <label>
              Title:
            </label>
            <span>
              test track 1
            </span>
          </div>
          <div
            class="stream_id"
          >
            <label>
              Stream ID:
            </label>
            <span>
              CuklIb9d3fI
            </span>
          </div>
        </div>
        <div
          class="stream_buttons"
        >
          <a
            data-testid="copy-original-url"
            href="#"
          >
            <i
              aria-hidden="true"
              class="linkify icon"
            />
          </a>
          <a
            href="#"
          >
            <i
              aria-hidden="true"
              class="refresh icon"
            />
          </a>
        </div>
      </div>
      <hr />
      <div
        class="queue_popup_buttons_container"
      >
        <a
          aria-label="Play this track now"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="play icon"
          />
           Play now
        </a>
        <div
          aria-expanded="false"
          class="ui dropdown nuclear dropdown popup_button left"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="divider text"
            role="alert"
          >
            Add to playlist
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition left"
          >
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="music icon"
              />
              test playlist
            </div>
          </div>
        </div>
        <a
          aria-label="Add this track to favorites"
          class="popup_button"
          data-testid="queue-popup-favorite"
          href="#"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
           Add to favorites
        </a>
        <a
          aria-label="Download this track"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="download icon"
          />
           Download
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Play Queue container should display a context popup on right click for tracks without thumbnails 1`] = `
<DocumentFragment>
  <div
    class="play_queue_container"
  >
    <div
      class="queue_menu_container"
    >
      <div
        class="queue_menu_buttons"
      >
        <a
          class="compactButton"
          data-testid="queue-menu-collapse"
          href="#"
        >
          <i
            aria-hidden="true"
            class="angle right icon"
          />
        </a>
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="ui item dropdown queue_menu_more"
          data-testid="queue-menu-more-container"
          role="listbox"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="ellipsis vertical icon"
          />
          <div
            class="menu transition"
          >
            <div
              class="header"
            >
              Queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="trash icon"
              />
              Clear queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="save icon"
              />
              Save as playlist
            </div>
            <div
              class="divider"
            />
            <div
              class="header"
            >
              Current track
            </div>
            <div
              class="item"
              role="option"
            >
              <div
                aria-expanded="false"
                class="ui dropdown left"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  class="divider text"
                  role="alert"
                >
                  Add to playlist
                </div>
                <i
                  aria-hidden="true"
                  class="dropdown icon"
                />
                <div
                  class="menu transition left playlists_menu"
                >
                  <div
                    class="item"
                    role="option"
                  >
                    <i
                      aria-hidden="true"
                      class="music icon"
                    />
                    test playlist
                  </div>
                </div>
              </div>
            </div>
            <div
              class="item"
              data-testid="queue-menu-more-favorite"
              role="option"
            >
              <i
                aria-hidden="true"
                class="star icon"
              />
              Add to favorites
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
              Download
            </div>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      class="play_queue_items fade_in"
      data-rbd-droppable-context-id="3"
      data-rbd-droppable-id="play_queue"
    >
      <div
        aria-describedby="rbd-hidden-text-3-hidden-text-15"
        data-rbd-drag-handle-context-id="3"
        data-rbd-drag-handle-draggable-id="uuid1+0"
        data-rbd-draggable-context-id="3"
        data-rbd-draggable-id="uuid1+0"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid1"
        >
          <div
            class="nuclear queue_item current_song"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 1
              </div>
              <div
                class="artist_container"
              >
                test artist 1
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-3-hidden-text-15"
        data-rbd-drag-handle-context-id="3"
        data-rbd-drag-handle-draggable-id="uuid2+1"
        data-rbd-draggable-context-id="3"
        data-rbd-draggable-id="uuid2+1"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid2"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 2
              </div>
              <div
                class="artist_container"
              >
                test artist 2
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-3-hidden-text-15"
        data-rbd-drag-handle-context-id="3"
        data-rbd-drag-handle-draggable-id="uuid3+2"
        data-rbd-draggable-context-id="3"
        data-rbd-draggable-id="uuid3+2"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid3"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="test-file-stub"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 3
              </div>
              <div
                class="artist_container"
              >
                test artist 3
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-3"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-3-hidden-text-15"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="ui bottom center popup transition visible queue_popup hidden"
      style="position: initial;"
    >
      <div
        class="stream_info"
      >
        <div
          class="stream_thumbnail"
        />
        <div
          class="stream_text_info"
        >
          <div
            class="stream_source"
          >
            <label>
              Source:
            </label>
             
            <div
              aria-expanded="false"
              class="ui inline dropdown"
              data-testid="stream-info-dropdown"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-atomic="true"
                aria-live="polite"
                class="divider text"
                role="alert"
              >
                Test Stream Provider
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              >
                <div
                  aria-checked="true"
                  aria-selected="true"
                  class="active selected item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    Test Stream Provider
                  </span>
                </div>
                <div
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    Different Stream Provider
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="stream_title"
          >
            <label>
              Title:
            </label>
            <span>
              test track 3
            </span>
          </div>
          <div
            class="stream_id"
          >
            <label>
              Stream ID:
            </label>
            <span>
              CuklIb9d3fI
            </span>
          </div>
        </div>
        <div
          class="stream_buttons"
        >
          <a
            href="#"
          >
            <i
              aria-hidden="true"
              class="refresh icon"
            />
          </a>
        </div>
      </div>
      <hr />
      <div
        class="queue_popup_buttons_container"
      >
        <a
          aria-label="Play this track now"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="play icon"
          />
           Play now
        </a>
        <div
          aria-expanded="false"
          class="ui dropdown nuclear dropdown popup_button left"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="divider text"
            role="alert"
          >
            Add to playlist
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition left"
          >
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="music icon"
              />
              test playlist
            </div>
          </div>
        </div>
        <a
          aria-label="Add this track to favorites"
          class="popup_button"
          data-testid="queue-popup-favorite"
          href="#"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
           Add to favorites
        </a>
        <a
          aria-label="Download this track"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="download icon"
          />
           Download
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Play Queue container should display tracks in the queue 1`] = `
<DocumentFragment>
  <div
    class="play_queue_container"
  >
    <div
      class="queue_menu_container"
    >
      <div
        class="queue_menu_buttons"
      >
        <a
          class="compactButton"
          data-testid="queue-menu-collapse"
          href="#"
        >
          <i
            aria-hidden="true"
            class="angle right icon"
          />
        </a>
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="ui item dropdown queue_menu_more"
          data-testid="queue-menu-more-container"
          role="listbox"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="ellipsis vertical icon"
          />
          <div
            class="menu transition"
          >
            <div
              class="header"
            >
              Queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="trash icon"
              />
              Clear queue
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="save icon"
              />
              Save as playlist
            </div>
            <div
              class="divider"
            />
            <div
              class="header"
            >
              Current track
            </div>
            <div
              class="item"
              role="option"
            >
              <div
                aria-expanded="false"
                class="ui dropdown left"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  class="divider text"
                  role="alert"
                >
                  Add to playlist
                </div>
                <i
                  aria-hidden="true"
                  class="dropdown icon"
                />
                <div
                  class="menu transition left playlists_menu"
                >
                  <div
                    class="item"
                    role="option"
                  >
                    <i
                      aria-hidden="true"
                      class="music icon"
                    />
                    test playlist
                  </div>
                </div>
              </div>
            </div>
            <div
              class="item"
              data-testid="queue-menu-more-favorite"
              role="option"
            >
              <i
                aria-hidden="true"
                class="star icon"
              />
              Add to favorites
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
              Download
            </div>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      class="play_queue_items fade_in"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="play_queue"
    >
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="uuid1+0"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="uuid1+0"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid1"
        >
          <div
            class="nuclear queue_item current_song"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 1
              </div>
              <div
                class="artist_container"
              >
                test artist 1
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="uuid2+1"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="uuid2+1"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid2"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="https://test-track-thumb-url"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 2
              </div>
              <div
                class="artist_container"
              >
                test artist 2
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="uuid3+2"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="uuid3+2"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          data-testid="queue-popup-uuid3"
        >
          <div
            class="nuclear queue_item"
          >
            <div
              class="thumbnail"
            >
              <img
                src="test-file-stub"
              />
              <div
                class="thumbnail_overlay"
              >
                <i
                  aria-hidden="true"
                  class="trash alternate outline big icon"
                />
              </div>
            </div>
            <div
              class="item_info_container"
            >
              <div
                class="name_container"
              >
                test track 3
              </div>
              <div
                class="artist_container"
              >
                test artist 3
              </div>
            </div>
            <div
              class="item_duration_container"
            >
              <div
                class="item_duration"
              >
                05:00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-0"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-0-hidden-text-0"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
</DocumentFragment>
`;
